### ISSUE TITLE
Buffer not send using the Ping class

### ISSUE THREAD

**Issue by nikswap** [dotnet/corefx#41033]
Using the Ping class Send method is the buffer not included in the call. This is found to be when it uses the built in ping command. 

The -p option is not populated in the method https://github.com/dotnet/corefx/blob/ecad65740b41b1502188ac6b412ade53fb54ba3e/src/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs#L73

**Comment by davidsh** [comment]
@wfurt

**Comment by wfurt** [comment]
There is really not good way how to do this unless executed as root when we have access to RAW socket. Different distributions have different options as well as there is size limit for -p option.
Perhaps we should improve documentation and  describe it as limitation.

**Comment by nikswap** [comment]
I think that most distros agrees on the -p option.

At least Debian based and Mac OS X both have the -p option.

However. In the documentation on Debian does it say the following:
```
-p pattern
           You may specify up to 16 “pad” bytes to fill out the packet you
           send. This is useful for diagnosing data-dependent problems in a
           network. For example, -p ff will cause the sent packet to be filled
           with all ones.
```
meaning that an error / message / something to tell the user that only 16 bytes can be used might also be an option?

**Comment by scalablecory** [comment]
I don't know if it would be worth it, but what is the complexity of discovering capabilities at runtime?

**Comment by wfurt** [comment]
we possibly could. The issue is that if we miss and pass unrecognized option, everything fails. There was some discussion about when we added TTL handling. 

Also this only solve sending part. When somebody want to fill body with specific pattern and then verify that response buffer has (in)proper content you cannot do that - unlike Windows and running on top of RAW socket. We can come up with improvements but I don't think we can make it 100% same as on Windows.

**Comment by nikswap** [comment]
True. Regarding the part of verifying the data sent. Now does it seem like
that the code sends the data but nothing is returned.

I am quite sure that the server do not need to return the data sent but can
choose to do anything.
